packages/protect/src/lib/protect.ts

```diff
@@ -0,0 +1,239 @@
+
+        if (options.behavioralDataCollection === true) {
+          window._pingOneSignals.resumeBehavioralData();
+        }
+      },
+      getData: async (): Promise<string> => {
+        return await window._pingOneSignals.getData();
+      },
+      pauseBehavioralData: (): void => {
+        window._pingOneSignals.pauseBehavioralData();
+      },
+      resumeBehavioralData: (): void => {
+        window._pingOneSignals.resumeBehavioralData();
+      },
+
+      /** ***********************************************************************************************
+       * Required when using the Ping Protect Marketplace nodes, which has generic callbacks
+       * But, can be used for native nodes and/or either callback type
+       */
+      getPauseBehavioralData: (step: FRStep): boolean => {
+        // Check for native callback first
+        try {
+          const nativeCallback = step.getCallbackOfType<PingOneProtectEvaluationCallback>(
+            CallbackType.PingOneProtectEvaluationCallback,
+          );
+
+          const shouldPause = nativeCallback?.getPauseBehavioralData();
+          return shouldPause || false;
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        } catch (err) {
+          // Do nothing
+        }
+
+        // If we are here, we are dealing with Marketplace nodes
+        const cbs = step.getCallbacksOfType<MetadataCallback>(CallbackType.MetadataCallback);
+
+        if (!cbs.length) {
+          return false;
+        }
+
+        const protectMetadataCb = cbs.find((metadataCallback: MetadataCallback) => {
+          const data = metadataCallback.getData() as { _type: string; _action: string };
+          return data._type === 'PingOneProtect';
+        });
+
+        if (!protectMetadataCb) {
+          return false;
+        }
+
+        const data: ProtectInitializeConfig | ProtectEvaluationConfig = (
+          protectMetadataCb as MetadataCallback
+        ).getData();
+
+        if (data._action === 'protect_risk_evaluation') {
+          return false;
+        } else {
+          return !!(data as ProtectInitializeConfig).behavioralDataCollection;
+        }
+      },
+      getNodeConfig: (step: FRStep): ProtectInitializeConfig | undefined => {
+        // Check for native callback first
+        try {
+          const nativeCallback = step.getCallbackOfType<PingOneProtectInitializeCallback>(
+            CallbackType.PingOneProtectInitializeCallback,
+          );
+
+          const config = nativeCallback?.getConfig() as ProtectInitializeConfig;
+          return config;
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        } catch (err) {
+          // Do nothing
+        }
+
+        const cbs = step.getCallbacksOfType<MetadataCallback>(CallbackType.MetadataCallback);
+
+        if (!cbs.length) {
+          return undefined;
+        }
+
+        const protectMetadataCb = cbs.find((metadataCallback) => {
```

Comment:
is there any chance that there are no `metadata` callbacks and this produces `undefined`?

Reply:
I don't remember, so maybe i should look into this. I maybe wrote it like this originally.

Reply:
nevermind I see the check below